src/slave/slave.cpp

```diff
@@ -3654,6 +3655,32 @@ void Slave::launchExecutor(
       authenticationToken.get(),
       framework->info.checkpoint());
 
+
+  if(environment.find("CRITEO_CNI") != environment.end() && environment["CRITEO_CNI"] == "enabled"){
```

Comment:
isn't it possible for the CNI plugin to do that change itself?

Reply:
if you mean injecting subnets then no, CNI plugins are called by the cni isolator for every network according to the config. If the container isn't part of any network then they won't be called in the first place.
Besides, according to the spec it's the runtime engine (ie mesos in this case)'s responsibility to identify which networks a container should join. CNI plugins are only responsible for making the necessary changes to ensure container connectivity.